### PR TITLE
fix: loading string was visible 

### LINKED ISSUE #1064: 🐛 [Bug report]: there is a "loading..." string that gets visible sometimes
### Descrcibe the bug?

On the reactplay.io/plays page, there is a "loading..." string that gets visible sometimes after navigating between reactplay.io/ and reactplay.io/plays pages a few times.

### Steps to reproduce the bug?

1. Go to reactplay.io
2. Navigate a few times between the root and /plays page
3. Note the "loading..." under the header

### Expected behavior

The "loading..." string should not be shown and the top banner should have been visible

### Desktop (Please provide your system information)

OS: Windows 11
Browser: Chrome 
Version: 111.0.5563.64

### Mobile (Please provide your device information)

_No response_

### Screenshot / Screenshare

Recording 2023-03-21 075328

### Relevant log output

```shell
none
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Mohitb07**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #1064

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

# Screenshots or example output

## Before
image

## After
image

### CODE REVIEW

**Comment by Mohitb07** [comment]
Review Ready @atapas

File: src/common/playlists/DynamicBanner.jsx [hunk 1]
```diff
@@ -3,41 +3,55 @@ import thumbPlay from 'images/thumb-play.png';
...
 
-          console.error(
-            `Cover image not found for the play ${randomPlay.name}. Setting the default cover image...`
-          );
+          if (coverImage) {
+            setCoverImage(coverImage);
+          } else {
+            setCoverImage(thumbPlay);
+
+            console.error(
```
Comment (joshi-kaushal) [inline src/common/playlists/DynamicBanner.jsx:27]: Rather than logging the error on the console, we can prompt errors with `alert` or display a toast. this would be a better user experience.

File: src/common/playlists/DynamicBanner.jsx [hunk 2]
```diff
@@ -3,41 +3,55 @@ import thumbPlay from 'images/thumb-play.png';
...
+              `Cover image not found for the play ${randomPlay.name}. Setting the default cover image...`
+            );
+          }
         }
+      } catch (error) {
+        console.error(`Error setting play cover: ${error}`);
```
Comment (joshi-kaushal) [inline src/common/playlists/DynamicBanner.jsx:33]: Again, toast or alert

File: src/common/playlists/DynamicBanner.jsx [hunk 3]
```diff
@@ -3,41 +3,55 @@ import thumbPlay from 'images/thumb-play.png';
...
 
     setPlayCover();
   }, [randomPlay]);
 
-  if (loading) return <p>loading...</p>;
+  if (loading) {
+    return (
+      <div
+        className="dynamic-banner-container"
+        style={{
```
Comment (joshi-kaushal) [inline src/common/playlists/DynamicBanner.jsx:46]: Inline CSS is usually a bad practice. Use Tailwind CSS Classes or a separate `.css` file instead.

File: src/common/playlists/PlayList.jsx [hunk 4]
```diff
@@ -58,7 +57,7 @@ const PlayList = () => {
     return <Loader />;
   }
 
-  if (plays?.length === 0) {
+  if (plays.length === 0) {
```
Comment (joshi-kaushal) [inline src/common/playlists/PlayList.jsx:60]: Can you help me understand why did you remove optional chaining (`?`) here? I am confused.

**Review by joshi-kaushal — DISMISSED** [review]

File: src/common/playlists/DynamicBanner.jsx (hunk 3 above)
Comment (Mohitb07) [reply src/common/playlists/DynamicBanner.jsx:46]: Yeah, I agree it's a bad practice but it was used in the same component's jsx where we are embedding dynamic banner from the state as a cover image directly therefore to show the same kind of placeholder I have used the same CSS style here.

File: src/common/playlists/DynamicBanner.jsx (hunk 1 above)
Comment (Mohitb07) [reply src/common/playlists/DynamicBanner.jsx:27]: Yeah, that would be better.

File: src/common/playlists/DynamicBanner.jsx (hunk 2 above)
Comment (Mohitb07) [reply src/common/playlists/DynamicBanner.jsx:33]: Sure but we have 2 errors so do we need two toasts for each where one is if something went wrong while fetching the banner and the other where if the banner is not found for the play?

**Comment by atapas** [comment]
@joshi-kaushal please finish your review

**Comment by atapas** [comment]
@Mohitb07 please resolve the merge conflicts

**Comment by Mohitb07** [comment]
> @Mohitb07 please resolve the merge conflicts

@atapas Done

File: src/common/playlists/DynamicBanner.jsx [hunk 5]
```diff
@@ -3,37 +3,59 @@ import thumbPlay from 'images/thumb-play.png';
...
-          );
+          // if it is not passed as a meta data
+          // check in the play folder for a cover image with the name cover
+          const coverImage = await loadCoverImage(randomPlay.slug);
+
+          if (coverImage) {
+            setCoverImage(coverImage);
+          } else {
+            toast.warning('Unable to find banner default set');
+            setCoverImage(thumbPlay);
```
Comment (joshi-kaushal) [inline src/common/playlists/DynamicBanner.jsx:27]: Where is `thumbPlay` import?

Comment (Mohitb07) [reply src/common/playlists/DynamicBanner.jsx]: fixed now

**Comment by priyankarpal** [comment]
merged https://github.com/reactplay/react-play/pull/1146

**Comment by Mohitb07** [comment]
> merged #1146

Hello, there have been some updates other than the loading screen issue that you may find helpful. Also, I noticed that the PR you merged doesn't display any errors to the user if the banner fails to load, which was a requirement.
